include/alpaka/intrinsic/IntrinsicUniformCudaHipBuiltIn.hpp

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2020 Sergei Bastrakov
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+
+#include <alpaka/core/BoostPredef.hpp>
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && !BOOST_LANG_CUDA
+    #error If ALPAKA_ACC_GPU_CUDA_ENABLED is set, the compiler has to support CUDA!
+#endif
+
+#if defined(ALPAKA_ACC_GPU_HIP_ENABLED) && !BOOST_LANG_HIP
+    #error If ALPAKA_ACC_GPU_HIP_ENABLED is set, the compiler has to support HIP!
+#endif
+
+#include <alpaka/intrinsic/Traits.hpp>
+
+namespace alpaka
+{
+    namespace intrinsic
+    {
+        //#############################################################################
+        //! The GPU CUDA/HIP intrinsic.
+        class IntrinsicUniformCudaHipBuiltIn : public concepts::Implements<ConceptIntrinsic, IntrinsicUniformCudaHipBuiltIn>
+        {
+        public:
+            //-----------------------------------------------------------------------------
+            IntrinsicUniformCudaHipBuiltIn() = default;
+            //-----------------------------------------------------------------------------
+            __device__ IntrinsicUniformCudaHipBuiltIn(IntrinsicUniformCudaHipBuiltIn const &) = delete;
+            //-----------------------------------------------------------------------------
+            __device__ IntrinsicUniformCudaHipBuiltIn(IntrinsicUniformCudaHipBuiltIn &&) = delete;
+            //-----------------------------------------------------------------------------
+            __device__ auto operator=(IntrinsicUniformCudaHipBuiltIn const &) -> IntrinsicUniformCudaHipBuiltIn & = delete;
+            //-----------------------------------------------------------------------------
+            __device__ auto operator=(IntrinsicUniformCudaHipBuiltIn &&) -> IntrinsicUniformCudaHipBuiltIn & = delete;
+            //-----------------------------------------------------------------------------
+            ~IntrinsicUniformCudaHipBuiltIn() = default;
+        };
+
+        namespace traits
+        {
+            //#############################################################################
+            template<>
+            struct Popcount<
+                IntrinsicUniformCudaHipBuiltIn>
+            {
+                //-----------------------------------------------------------------------------
+                __device__ static auto popcount(
+                    intrinsic::IntrinsicUniformCudaHipBuiltIn const & /*intrinsic*/,
+                    unsigned int value)
+                -> int
+                {
+                    return __popc(value);
```

Comment:
Oh no, clang + CUDA for some reason has the input as `int`, not `unsigned` like every other implementation.

Reply:
This causes tests to fail. Will fix